chain/datasource/datasource.go

```diff
@@ -192,13 +188,30 @@ func (t *DataSource) Store() adt.Store {
 }
 
 func (t *DataSource) Actor(ctx context.Context, addr address.Address, tsk types.TipSetKey) (*types.Actor, error) {
+	metrics.RecordInc(ctx, metrics.DataSourceActorCacheRead)
 	ctx, span := otel.Tracer("").Start(ctx, "DataSource.Actor")
 	if span.IsRecording() {
 		span.SetAttributes(attribute.String("tipset", tsk.String()))
 		span.SetAttributes(attribute.String("address", addr.String()))
 	}
 	defer span.End()
-	return t.node.StateGetActor(ctx, addr, tsk)
+
+	key, err := asKey(addr, tsk)
+	if err != nil {
+		return nil, err
```

Comment:
not sure when this would happen, but shouldn't we still try to continue and call `t.node.StateGetActor` instead of returning error?
```
if err == nil {
  value, found := t.actorCache.Get(key)
  ...
}
```

Reply:
fixed!